recognition/scoring.py

```diff
@@ -12,7 +12,7 @@
 import tempfile
 import collections
 import re
-from typing import List, Optional
+from typing import List, Optional, Union
```

Comment:
```suggestion
from typing import List, Optional
```
If we disallow `str` below, then we don't need `Union`